src/EmbeddedVsix.UnitTests/TypeScriptRuleDefinitions_SanityChecks.cs

```diff
@@ -45,12 +45,12 @@ public void LoadRuleMetadataFromFile()
             File.Exists(filePath).Should().BeTrue("Test setup error: could not find rule metadata file. Expected path: " + filePath);
 
             // Sanity check that the json file is loadable and has rules
-            var rulesProvider = new RulesRepository(filePath);
+            var factory = new RulesProviderFactory(filePath);
```

Comment:
Miscellaneous: changed the sanity check test to use the new factory.

As part of this work I had deleted originally _RulesRepository_ and the associated tests, which forced me to fix this up. However, I can't completely remove _RulesRepository_ yet as the active rules stuff still references the old IJavaScriptRulesDefinitionProvider interface, so I re-instated it for the time being.